File: HotDesk_PRAI/Backup/Library/Library.Root/TemplateField/checkboxfield.vb
Imports System.Web.UI.WebControls

Public Class checkboxfield
    Implements System.Web.UI.ITemplate

    Public Const CheckboxHeaderID As String = "chkall"
    Public Const CheckboxItemID As String = "ckitem"
    Private plittype As ListItemType

    Public Sub New(ByVal type As ListItemType)
        MyBase.New()
        Me.plittype = type
    End Sub

    Public Sub InstantiateIn(ByVal container As System.Web.UI.Control) Implements System.Web.UI.ITemplate.InstantiateIn
        Dim ckitem As CheckBox
        Dim ckall As CheckBox

        Select Case Me.plittype
            Case listItemType.Header
                ckall = New CheckBox()
                ckall.ID = "chkall"
                ckall.AutoPostBack = True
                container.Controls.Add(ckall)
            Case listItemType.Footer
                Return
            Case listItemType.Item
                ckitem = New CheckBox()
                ckitem.ID = "ckitem"
                ckitem.AutoPostBack = True
                container.Controls.Add(ckitem)
        End Select
    End Sub
End Class

File: HotDesk_PRAI/Library/Library.Root/TemplateField/checkboxfield.cs
using System.Web.UI.WebControls;

public class checkboxfield : System.Web.UI.ITemplate
{
    public const string CheckboxHeaderID = "chkall";
    public const string CheckboxItemID = "ckitem";
    private ListItemType plittype;

    public checkboxfield(ListItemType type)
    {
        this.plittype = type;
    }

    public void InstantiateIn(System.Web.UI.Control container)
    {
        CheckBox ckitem;
        CheckBox ckall;

        switch (this.plittype)
        {
            case ListItemType.Header:
                ckall = new CheckBox();
                ckall.ID = "chkall";
                ckall.AutoPostBack = true;
                container.Controls.Add(ckall);
                break;
            case ListItemType.Footer:
                return;
            case ListItemType.Item:
                ckitem = new CheckBox();
                ckitem.ID = "ckitem";
                ckitem.AutoPostBack = true;
                container.Controls.Add(ckitem);
                break;
        }
    }
}
